drivers/gpio/gpio_tca6424a.c

```diff
@@ -0,0 +1,610 @@
+ * The interrupt is configured if it is enabled.
+ *
+ * @param dev Device struct
+ *
+ * @retval 0 if successful.
+ * @retval Negative value for error code.
+ */
+static int tca6424a_init(const struct device *dev)
+{
+	const struct tca6424a_drv_cfg *drv_cfg = dev->config;
+	struct tca6424a_drv_data *drv_data = dev->data;
+	int ret = 0;
+
+	if (!device_is_ready(drv_cfg->i2c_spec.bus)) {
+		LOG_ERR("I2C device not found");
+		return -ENODEV;
+	}
+	/* If the RESET line is available, use it to reset the expander.
+	 * Otherwise, write reset values to registers that are not used by
+	 * this driver.
+	 */
+	if (drv_cfg->reset_gpio.port) {
+		if (!device_is_ready(drv_cfg->reset_gpio.port)) {
+			LOG_ERR("%s is not ready", drv_cfg->reset_gpio.port->name);
+			return -ENODEV;
+		}
+		ret = gpio_pin_configure_dt(&drv_cfg->reset_gpio,
+				GPIO_OUTPUT_ACTIVE);
+		if (ret != 0) {
+			LOG_ERR("%s: failed to configure RESET line: %d",
+				dev->name, ret);
+			return -EIO;
+		}
+		/* RESET signal needs to be active for a minimum of 30 ns. */
+		k_busy_wait(1);
+
+		ret = gpio_pin_set_dt(&drv_cfg->reset_gpio, 0);
+		if (ret != 0) {
+			LOG_ERR("%s: failed to deactivate RESET line: %d",
+				dev->name, ret);
+			return -EIO;
+		}
+		/* Give the expander at least 200 ns to recover after reset. */
+		k_busy_wait(1);
+	} else {
+		ret = update_invers_regs(dev, 0x0);
+		if (ret != 0) {
+			LOG_ERR("%s: failed to reset inversion register: %d",
+				dev->name, ret);
+			return -EIO;
+		}
+	}
+	/* Set initial configuration of the pins. */
+	ret = update_config_regs(dev, 0xFFFFFF);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = update_output_regs(dev, 0x0);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Read initial state of the input port register. */
+	ret = update_input_regs(dev, &drv_data->pins_state.input);	
+	if (ret != 0) {
+		LOG_ERR("%s: failed to initially read input port: %d",
+			dev->name, ret);
+		return -EIO;
+	}
+
+	/* If the INT line is available, configure the callback for it. */
+	if (drv_cfg->int_gpio.port) {
+		if (!device_is_ready(drv_cfg->int_gpio.port)) {
+			LOG_ERR("Cannot get pointer to gpio interrupt device");
+			LOG_ERR("%s init failed: %d", dev->name, -EINVAL);
+			return -EINVAL;
+		}
+
+		drv_data->dev = dev;
+	
+		k_work_init(&drv_data->work, tca6424a_work_handler);
+
+		ret = gpio_pin_configure_dt(&drv_cfg->int_gpio, GPIO_INPUT);
+		if (ret) {
+			LOG_ERR("%s init failed: %d", dev->name, ret);
+			return ret;
+		}
+
+		ret = gpio_pin_interrupt_configure_dt(&drv_cfg->int_gpio,
+						GPIO_INT_EDGE_TO_ACTIVE);
+		if (ret) {
+			LOG_ERR("%s init failed: %d", dev->name, ret);
+			return ret;
+		}
+
+		gpio_init_callback(&drv_data->int_gpio_cb,
+					tca6424a_int_gpio_handler,
+					BIT(drv_cfg->int_gpio.pin));
+
+		ret = gpio_add_callback(drv_cfg->int_gpio.port,
+					&drv_data->int_gpio_cb);
+		if (ret) {
+			LOG_ERR("%s init failed: %d", dev->name, ret);
+			return ret;
+		}
+	}	
+
+	LOG_INF("%s init ok", dev->name);
```

Comment:
I think LOG_DBG would be more appropriate here

Reply:
This call has been updated to LOG_DBG.